File: python_code/oop_class/house.py
# @version : 1.0
# @Author  : 混江龙
# @File    : house.py
# @Time    : 2025/9/2 14:08
"""
房屋数据对象
"""


class House:

    # 重新魔术方法__str__
    def __str__(self):
        return f"{self.id}\t\t{self.name}\t\t{self.phone}\t\t{self.address}\t\t{self.rent}\t\t{self.state}"

    def __init__(self, id, name, phone, address, rent, state):
        self.name = name
        self.phone = phone
        self.address = address
        self.rent = rent
        self.state = state
        self.id = id

File: java_code/houseRent/src/model/House.java
package model;

public class House {
    private int id;
    private String name;
    private String address;
    private String phone;
    private double rent;
    private String status;

    public House(int id, String name, String address, String phone, double rent, String status) {
        this.id = id;
        this.name = name;
        this.address = address;
        this.phone = phone;
        this.rent = rent;
        this.status = status;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public double getRent() {
        return rent;
    }

    public void setRent(double rent) {
        this.rent = rent;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    @java.lang.Override
    public String toString() {
        return id +
                "\t\t" + name +
                "\t\t" + address +
                "\t\t" + phone +
                "\t\t" + rent +
                "\t\t" + status ;
    }
}